misc/LADYBUG_USER_STORIES.md

```diff
@@ -3,51 +3,152 @@ Ladybug User Stories
+* [I want to configure whether my Frank application does produce reports](#i-want-to-configure-whether-my-frank-application-does-produce-reports)
+* [I do not want unauthorized access to reports](#i-do-not-want-unauthorized-access-to-reports)
+* [Miscelaneous](#miscelaneous)
+
+# I want to find the report I am interested in
+
+**10:** As a Frank tester, I want to see a table of reports when I open the Ladybug GUI. Each row of the table should be a report and each column should be a metadata attribute. In this table I can search the report of the adapter run that I want to examine.
+
+**20:** As a Frank tester, when I click a row in the table I want to see the corresponding report in the tree view. This allows me to examine it in more detail.
+
+**100:** As a Frank tester, I want to see at least the following metadata attributes in the report table:
+
+* **Name:** The name of the report, which is the name of the outermost start checkpoint.
+* **Correlation id:** See vision.
+* **Storage id:** The id assigned to the report when it was stored.
+* **End time:** The timestamp at which the outermost end checkpoint was created.
+* **Duration:** The time interval between the outermost start checkpoint and the outermost end checkpoint.
+* **Status:** The status with which the pipeline ended, should be SUCCESS or ERROR.
+* **Number of checkpoints**
+* **Estimated memory usage:** TODO: What does this mean?
+* **Storage size:** TODO: What does this mean?
+
+**120:** As a Frank developer, I want to be able to add columns to the report table in addition to the ones shown in story **100**. I am willing to write Spring XML files to achieve this.
+
+TODO: Do I have to specify what metadata extractors we have?
+
+**200:** As a Frank tester I want to hide reports from the report table in which I am not interested. This way I can find the report I need more easily. For each metadata field I want an edit box in which I can write a regular expression. Only if for a report all the metadata fields satisfy their regex, then the report should be shown in the table.
+
+TODO: What syntax do we support for the filters?
+
+**210:** As a Frank tester, I want to see the reports ordered by the timestamp of the outer start checkpoint or by the timestamp of the outer end checkpoint. The latest timestamp should be in the topmost table row.
+
+**300:** As a Frank tester I want to see how many reports-in-progress there are. There is a report-in-progress if for some correlation id the start checkpoint is not yet matched by a corresponding end checkpoint.
+
+**310:** As a Frank tester, if a report-in-progress is open for too long, then I want Ladybug to close it such that I can see it as a regular report in the table.
+
+**330:** As a Frank developer, I want to be able to configure the time threshold of story **310**.
+
+**400:** As a Frank tester, I want reports to be stored persistently. Reports should not vanish when the Frank!Framework is restarted.
```

Comment:
this is deployment method depending. containerized , the restart clears history

Reply:
Hebben we daarmee een user story voor de system administrator? Moet de system administrator kunnen bepalen wat voor storage we gebruiken, database, in-memory of filesystem?

Reply:
Ik heb iets opgenomen over de keuze van de storage.